packages/react-components/src/components/TextFieldWithMention/mentionTagHelpers.ts

```diff
@@ -0,0 +1,957 @@
+          });
+          processedChange = '';
+          result += stringBefore + updatedContent.updatedHTML;
+        } else {
+          // no subtags
+          result +=
+            htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex + tag.openTagBody.length) + processedChange;
+          processedChange = '';
+          // oldPlainTextEndIndex already includes mentionTag length
+          const endChangeDiff = closingTagInfo.plainTextEndIndex - oldPlainTextEndIndex;
+          // as change may be before the end of the tag, we need to add the rest of the tag
+          lastProcessedHTMLIndex = closingTagInfo.closeTagIndex - endChangeDiff;
+        }
+        // the change is handled; exit
+        break;
+      }
+      lastProcessedPlainTextTagEndIndex = closingTagInfo.plainTextEndIndex;
+    }
+
+    if (i === tags.length - 1 && oldPlainTextEndIndex >= closingTagInfo.plainTextEndIndex) {
+      // the last tag should handle the end of the change if needed
+      // oldPlainTextEndIndex already includes mentionTag length
+      const endChangeDiff = oldPlainTextEndIndex - closingTagInfo.plainTextEndIndex;
+      if (startIndex >= closingTagInfo.plainTextEndIndex) {
+        const startChangeDiff = startIndex - closingTagInfo.plainTextEndIndex;
+        result +=
+          htmlText.substring(
+            lastProcessedHTMLIndex,
+            closingTagInfo.closeTagIndex + closingTagInfo.closeTagLength + startChangeDiff
+          ) + processedChange;
+      } else {
+        result +=
+          htmlText.substring(lastProcessedHTMLIndex, closingTagInfo.closeTagIndex + closingTagInfo.closeTagLength) +
+          processedChange;
+      }
+      processedChange = '';
+      lastProcessedHTMLIndex = closingTagInfo.closeTagIndex + closingTagInfo.closeTagLength + endChangeDiff;
+      // the change is handled; exit
+      // break is not required here as this is the last element but added for consistency
+      break;
+    }
+  }
+  if (lastProcessedHTMLIndex < htmlText.length) {
+    // add the rest of the html string
+    result += htmlText.substring(lastProcessedHTMLIndex);
+  }
+  return { updatedHTML: result, updatedSelectionIndex: changeNewEndIndex };
+};
+
+/**
+ * Props for finding strings diff indexes
+ *
+ * @private
+ */
+export type DiffIndexesProps = {
+  // the old text
+  oldText: string;
+  // the new text
+  newText: string;
+  // the start of previous selection, should be a valid position in the input field
+  previousSelectionStart: number;
+  // the end of previous selection, should be a valid position in the input field
+  previousSelectionEnd: number;
+  // the start of current selection, should be a valid position in the input field
+  currentSelectionStart: number;
+  // the end of current selection, should be a valid position in the input field
+  currentSelectionEnd: number;
+};
+
+/**
+ * Result of finding strings diff indexes function
+ *
+ * @private
+ */
+export type DiffIndexesResult = {
```

Comment:
`IndexChanges`?

Reply:
it may be better to use`ChangeIndexes` as it's used to store info about indexes connected to a change